### DISCUSSION TITLE
Clarification for new "How can cars park here" quest?

### DISCUSSION THREAD

**Discussion by mnalis** [streetcomplete/StreetComplete#3606] [Q&A]
3f32f2086a6f1f3750de539afec62573ed564051 and thereabouts (to be included in `39.0`?) created a new quest `quest_street_parking_title`: "How do cars park here, if at all?"

I'm currently playing with it, but have encountered a few issues. See https://www.openstreetmap.org/note/2974100 for example. 

- looking at OSM wiki for [parking:lane:*](https://wiki.openstreetmap.org/wiki/Key:parking:lane), it seems to be intended for legally allowed parking spaces. Still, StreetComplete (same as OSM wiki) uses `can` instead of `may` (or even more formal ways) to explicitly involve _legality_ of the said parking, instead of _technical ability_  to park there.

  This leads to uncertainty. I do not know about statistics in more ordered countries, but in Croatia (and at least some of our eastern european neighbours), people park cars in **_way more_** places than where it is allowed (like, I would guess, at least 20%  of all cars are parked illegally at any time). 

  <details>

  Sometimes (when someone gets frustrated enough to insist on law being followed and footway being cleared) it results in illegally parked cars being towed away, but in majority of cases, it is just an inconvenient fact of life (like politicians being bribed/corrupt, world being overpolluted with plastics, etc). 

  For example, [legally in Croatia](https://zakon.hr/z/78/Zakon-o-sigurnosti-prometa-na-cestama), it is never allowed to park a car (fully or partially) at sidewalk _unless there is explicit sign for parking_ (which is not there in 99% of cases). It is however allowed to temporarily pull up at sidewalk (and stay there up to 3 minutes), but only if driver leaves at least 1.6m of free space for pedestrians. That is what is legally allowed, and what (I think) this tag is probably intended for.

  What instead happens in reality, is that people (illegally) park cars on sidewalks (blocking them fully) in majority of all residential streets. For example, in picture attached to that [note 2974100](https://www.openstreetmap.org/note/2974100): 
![72754](https://user-images.githubusercontent.com/156656/147290434-d2adfd42-9b3b-4de6-a6e4-c753a59e693b.jpg)

  the cars parked on the left side are legally allowed (marked) parking places, while on the right side is illegal parking (which takes whole of the pedestrian sidewalk).

  </details>

- Also, when one chooses `No parking here / Not allowed`, one is offered submenu with choices:

  ![small_Screenshot_20211223_220050_de westnordost streetcomplete mn debug](https://user-images.githubusercontent.com/156656/147291142-ff7b93da-f3f0-4ed7-b2d7-afe2904da6ac.jpg)
  The last one (`No explicit sign / parking not possible`) seems to tag with `parking:lane:right=no`, so it should also probably say `not allowed` instead of `not possible` (as we can clearly see in the picture in that note, it is clearly _possible_ - but it is also _illegal_). Or maybe adding another choice like `it is possible but not allowed`?

TL;DR - am I correct in my assumption that this quest is about _**legality**_ of parking at that place, instead of **_physical possibility_** of putting car there? And if so, what wordings would you suggest (especially as I assume situations/laws are not same in all countries)?

**Comment by mnalis** [comment]
Does anybody have a comment (or even see this at all :smiling_face_with_tear:)? 

@smichel17 do you have any language suggestions for better conveying the information that the quest is about legality of parking there (and not just a physical possibility)?

**Reply by matkoniecz** [reply]
I have seen some time ago - just had no good idea how to answer and how to change it.

**Reply by smichel17** [reply]
Same. Also, with the discussion of specific tag values it seemed the kind of thing that I usually ignore— as I've said before, as a coder I try to focus just on the android side, to preserve my perspective as an OSM-outsider (less successfully, as of late).

**Reply by smichel17** [reply]
> @smichel17 do you have any language suggestions for better conveying the information that the quest is about legality of parking there (and not just a physical possibility)?

You could say "How are cars [allowed|permitted] to park here?".

I recall a pull request where we discussed similar phrasing, probably any of the suggestions there would work. I think it was the one about parking lots being open to the general public vs customers only vs private.

**Comment by matkoniecz** [comment]
> but in Croatia (and at least some of our eastern european neighbours), people park cars in way more places than where it is allowed (like, I would guess, at least 20% of all cars are parked illegally at any time).

I can confirm this for Poland. 20% is pure guessing but seems likely.

> TL;DR - am I correct in my assumption that this quest is about legality of parking at that place, instead of physical possibility of putting car there? And if so, what wordings would you suggest (especially as I assume situations/laws are not same in all countries)?

It seems that it should be about legality, but in some cases the illegal parking is so ignored that...

**Comment by westnordost** [comment]
It is not defined in the wiki. So depending on who you ask, you will likely get different answers. I suppose that many OSM users, at least those that are louder, are very much focussed on the legal situation only. If you ask me, map what is on the ground. If there is de-facto no sidewalk because it is de-facto a parking lane, then it should be tagged that way.

There is written law and then there is how the law is enforced. If it is not, it is de-facto not a law. If the muncipal government really cared about that there is de-facto no sidewalk now, they'd just put up some bollards to hinder cars parking there or ticket those drivers.

**Reply by westnordost** [reply]
By the way, I frequently see no-stopping signs and people park anyway, just not on the roadway but on the sidewalk (beyond the curb). As far as I know, this is legal, as the no stopping sign only refers to the roadway. At least as long as the sidewalk is not obstructed by the car, i.e. there is at least X meters in width left. Not sure what the exact minimum was, I think it was just broad enough for a wide baby stroller.

**Reply by matkoniecz** [reply]
The worst case is where enforcement is occasional. For example fines are handed out once a year or once a month.

So it is neither really safe to park but still de facto parking lane.

**Reply by Dimitar5555** [reply]
> There is written law and then there is how the law is enforced. If it is not, it is de-facto not a law. If the muncipal government really cared about that there is de-facto no sidewalk now, they'd just put up some bollards to hinder cars parking there or ticket those drivers.

The law is there to be followed. The police/municipality can't cover the whole city in bollards nor it can check every place and fine all violations. This is why citizens should inform them about such problems and require action from them.


> By the way, I frequently see no-stopping signs and people park anyway, just not on the roadway but on the sidewalk (beyond the curb). As far as I know, this is legal, as the no stopping sign only refers to the roadway. At least as long as the sidewalk is not obstructed by the car, i.e. there is at least X meters in width left. Not sure what the exact minimum was, I think it was just broad enough for a wide baby stroller.

While it can be legal, sidewalks are usually not made strong enough to handle car's weight. A few years of such parking can cause a lot of damage which affects mainly the pedestrians. I'm on the opinion that OSM shouldn't stimulate actions that are against others or that ruin public or private property so if a sidewalk is used as a parking lane, it shouldn't be mapped as such and should be reported to the local authorities.

**Reply by matkoniecz** [reply]
> should

The problem is that in some places local government is not enforcing law and ignoring any requests to do so.

**Reply by Dimitar5555** [reply]
> > should
> 
> The problem is that in some places local government is not enforcing law and ignoring any requests to do so.

The problem could be in the way that you inform them. Writing official complaints to them usually works. If that doesn't, you can try sending it directly to the local police authority. If both of them don't work, the regional ombudsman or the regional public mediator can also help (assuming there are such people in your country).
Normally it should be defined in the laws what you should do if your complaint is not answered.

**Reply by matkoniecz** [reply]
> Writing official complaints to them usually works.

Oh, situation improved and nowadays they actually respond to emails.

About 8 years ago it was necessary to write official complaint about exceeding legally mandated time to respond to the actual official complaint. And typical answer was several pages explaining why nothing can be done, will be done or should be done.

But there are still places across world where things you mention exist only in theory.

**Reply by mnalis** [reply]
> It is not defined in the wiki

Well, perhaps not in so many words, but many of the key values don't make sense at all if tag was about _physical possibility_ instead of _legality_:

<details>

(eg. `no_parking`, `no_stopping`, `no_standing` are physically the same, and only differ in their legality; also `fire_lane` is just a lane - it is only about legality of stopping there, etc). At least it seems so to me. So I assume the `can` on that wiki is used in the same way as `can` on [oneway=yes](https://wiki.openstreetmap.org/wiki/Key:oneway) wiki page - meaning "legally allowed".

It is likewise with `fee=yes` on `amenity=parking` for example - yeah, you can sometimes (even often in some cases) park there without paying even if payment is required by law, but it does not mean one should mark as `fee=no` those spots where you're (noticably) less likely to get caught while breaking the law.  And most certainly StreetComplete should not use wording to suggest doing so is OK (IMHO).

</details>

> I suppose that many OSM users, at least those that are louder, are very much focussed on the legal situation only. If you ask me, map what is on the ground. If there is de-facto no sidewalk because it is de-facto a parking lane, then it should be tagged that way.

Well, people will always map what they *want*, we can't stop them there. But we can make the wording clearer, and in such a way that it does not promote illegal activities (as @Dimitar5555 notes).

As for mapping what is on the ground - I would always try to map tricky situation on the ground on _do-minimum-damage-on-wrong-or-partial-information_ basis:

<details>

e.g. in your example @westnordost - yes, I would map it as `sidewalk=no`, as I don't want some pedestrian counting on sidewalk being there, and then being hit by a car because it was clearly unusable as sidewalk, and my mapping being responsible for that. But I would **also** map it as  `parking:lane:both=no`, as **it is** illegal (even if not always enforced), and I do not want someone to park there and then have their car towed / fined  because they relied of the feature I mapped.

</details>

> There is written law and then there is how the law is enforced. If it is not, it is de-facto not a law.

Yes, but as @matkoniecz notes the law **_is enforced_** (at least here in Croatia), just _not regularly enough_ to effectively deter majority of car drivers from breaking it. I.e. it is type of lottery - you physically can park there if you can't find any other spot and you really need to park it now, but there is (say) 2% chance you won't find your car when you come back, and will have to go to municipality and pay a fine in order to get it back.

> If the muncipal government really cared about that there is de-facto no sidewalk now, they'd just put up some bollards to hinder cars parking there or ticket those drivers.



<details> 

<summary>I wish it was that simple.</summary>

I actually feel their position is, as saying goes, between anvil and a hammer. 

If they don't put bollards there, they'll have an **angry mob of pedestrians** who can't even send children to school safely because of car drivers breaking the law. If they however do put up the bollards, many of the people living in those apartments in the picture above would either have to park their cars at least several kilometers away from their homes (in which case they'll have an **angry mob of car drivers** at their throats who would also shortly likely destroy those bollards, thus reverting to the angry mob of pedestrians).  

</details>

---

**TL;DR** people will always answer as they wish, but SC should at least make clear what is recommended way by using clear wording, e.g.:

- `quest_street_parking_title` = `How are cars permitted to park here, if at all?`
- `street_parking_no_other_reasons` = `No explicit sign / parking not allowed`

**Comment by smichel17** [comment]
I have a similar (same?) question: How do I answer in the following situation?

- It is not marked whether cars are are *allowed* to park there.
    - There are no marked parking spots.
    - There is no "no parking" sign.
- I think it is legal to park here, but I am **not sure**.
- In any case, **in practice** you are certainly not going to get a ticket for parking here. It is a somewhat rural area and these are residential streets with almost no cars parked on the street (they are in driveways). Not like some city or suburb where parking is scarce and cops go around giving tickets to people without a permit.

The right answer might be "Parallel > On roadway". However, I wasn't sure if this was right, because I'm not sure what the **legal** situation is.

**Reply by mnalis** [reply]
Well, when I'm not sure what the exact situation on the ground is, I usually just select `Other answers / Can't say` (and if it happens in overwhelming number of cases, I just deselect that quest in my default preset). But maybe that is just me, as I'm very averse to giving **wrong** answers.

> In any case, in practice you are certainly not going to get a ticket for parking here

When you say "`certainly`" here, do you mean it as in mathematical proof of `100.0%` (e.g. it never ever happened even without you knowing, and you'd pledge to pay the fine from your own pocket for every individual that ever gets fined at that spot), or more like "very rarely" (ie. in `99.9%` of the cases nobody gets fined, and you've never seen anybody fined, but if they do get fined, that was just their bad luck)?

As I said in the discussion before, I personally approach it from **_minimize-worst-case-scenario_** angle (and would recommend it to everybody :smile:) :
- if I said `parking=no`, and I was wrong, the worst that would happen is that someone who could have found closer parking would have to look some more - basically the same as if the feature was not mapped (which happens a lot all the time). At worst they'd be "oh well why wasn't this parking shown" and maybe one of them would eventually correct my mistake or leave a note leading to fixing of mistake.
- if I however said `parking=yes`, and I was wrong, the worst that would happen is that someone (or multiple of them) would park there and get a fine, and would get very angry.

I prefer **not** to make people angry. So, I'd skip the quest (or, if I was forced to answer it, I'd rather say `parking=no`). And if it was really bothering me, I'd go and read what a law says so I'd be sure next time (for reference, in Croatia, unless it is signed, it is always illegal to park on sidewalk, but it can still be used for shorter than 3 minutes stay _if_ enough space is left for pedestrians)
